crates/aggchain-proof-builder/src/lib.rs

```diff
@@ -61,6 +63,27 @@ pub struct AggchainProofBuilderRequest {
     pub ger_inclusion_proofs: HashMap<String, InclusionProof>,
 }
 
+impl AggchainProofBuilderRequest {
+    /// Extract the proof and check the vkey is the expected one.
+    fn agg_span_proof_vkey(&self, expected_vkey_hash: B256) -> Result<SP1VerifyingKey, Error> {
```

Comment:
```suggestion
    fn check_aggspan_proof_vkey(&self, expected_vkey_hash: B256) -> Result<(), Error> {
```

Here, we only care about knowing if the vKey of the proof matches the one in our config.